File: src/app/core/components/staff/model/staff.ts
import { FormControl } from "@angular/forms";
import { Department, IDepartment } from "../../company/model/department";
import { IUser, User } from "../../user/model/user";
import { IOccupation, Occupation } from "./occupation";

export interface IStaff {
    id: string;
    fullName: string;
    email: string;
    department: IDepartment;
    occupation: IOccupation;
    user: IUser;
    active: boolean;
}

export class Staff implements IStaff {
    id: string;
    fullName: string;
    email: string;
    department: IDepartment;
    occupation: IOccupation;
    user: IUser;
    active: boolean;

    constructor(id: string, fullName: string, email: string, department: IDepartment, occupation: IOccupation
        , user: User, active: boolean) {
        this.id = id;
        this.fullName = fullName;
        this.email = email;
        this.department = department;
        this.occupation = occupation;
        this.user = user;
        this.active = active;
    }
}

export interface StaffForm {
    fullName: FormControl<string>;
    email: FormControl<string>;
    department: FormControl<Department>;
    occupation: FormControl<Occupation>;
    active: FormControl<boolean>;
}

File: src/main/java/io/github/mroncatto/itflow/domain/staff/entity/Staff.java
package io.github.mroncatto.itflow.domain.staff.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import io.github.mroncatto.itflow.domain.company.entity.Department;
import io.github.mroncatto.itflow.domain.user.entity.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.io.Serial;
import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

@Entity
@Table
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Staff implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(length = 75, nullable = false)
    private String fullName;

    @Column(length = 45)
    private String email;

    @ManyToOne(optional = false)
    private Department department;

    @ManyToOne(optional = false)
    private Occupation occupation;

    @OneToOne(mappedBy = "staff")
    @JsonIgnoreProperties("staff")
    private User user;

    @Column(nullable = false)
    private boolean active;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Staff staff = (Staff) o;
        return id != null && Objects.equals(id, staff.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
